mysite/views.py

```diff
@@ -1,3 +1,175 @@
+
+
+def project_view(request, pk, template_name='mysite/project_view.html'):
+    project = get_object_or_404(Project, pk=pk)
+    return render(request, template_name, {'project': project})
+
+
+def project_update(request, pk, template_name='mysite/project_form.html'):
+    project = get_object_or_404(Project, pk=pk)
+    form = ProjectForm(request.POST or None, instance=project)
+    if form.is_valid():
+        form.save()
+        return redirect('project_list')
+    return render(request, template_name, {'form': form})
+
+
+def project_delete(request, pk, template_name='mysite/project_confirm_delete.html'):
+    project = get_object_or_404(Project, pk=pk)
+    if request.method == 'POST':
+        project.delete()
+        return redirect('project_list')
+    return render(request, template_name, {'object': project})
+
+
+def project_bid_list(request, template_name='mysite/project_bid_list.html'):
+    project_bids = ProjectBid.objects.all()
+    data = {'object_list': project_bids}
+    return render(request, template_name, data)
+
+
+def project_bid_create(request, project_id, template_name='mysite/project_bid_form.html'):
+
+    form = ProjectBidForm(request.POST or None)
+    projects = Project.objects.all()
+    if form.is_valid():
+        project_bid = form.save(commit=False)
+        project_bid.bid_by = request.user
+        project_bid.project = projects.get(pk=project_id)
+        form.save()
+        return redirect('project_bid_list')
+    return render(request, template_name, {'form': form})
+
+
+def project_bid_update(request, pk, template_name='mysite/project_bid_form.html'):
+    project_bid = get_object_or_404(ProjectBid, pk=pk)
+    form = ProjectBidForm(request.POST or None, instance=project_bid)
+    if form.is_valid():
+        form.save()
+        return redirect('project_bid_list')
+    return render(request, template_name, {'form': form})
+
+
+def project_bid_delete(request, pk, template_name='mysite/project_bid_confirm_delete.html'):
+    project_bid = get_object_or_404(ProjectBid, pk=pk)
+    if request.method == 'POST':
+        project_bid.delete()
+        return redirect('project_bid_list')
+    return render(request, template_name, {'object': project_bid})
+
+
+def project_assign(request, pk, template_name='mysite/project_assign_form.html'):
+    project = get_object_or_404(Project, pk=pk)
+    user_ids = ProjectBid.objects.all().filter(project=project).values_list('bid_by', flat=True)
+    users = User.objects.filter(pk__in=list(user_ids))
+
+    form = AssignProjectForm(request.POST or None, initial={'users_list': list(users.values_list('id', 'username'))
+})
+    if form.is_valid():
+        form_data = form.cleaned_data
+        user_id = form_data['selected_user']
+        user = get_object_or_404(User, pk=user_id)
+        project.assigned_to = user.username
+        project.status = 'Assigned'
+        project.save()
+        return redirect('project_list')
```

Comment:
Move this post logic to form.save if you can.

Reply:
Feedback Incorporated.